validators/crossingHighways/map.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+var turf = require('turf');
+var _ = require('underscore');
+var rbush = require('rbush');
+
+module.exports = function(tileLayers, tile, writeData, done) {
+  var layer = tileLayers.osm.osm;
+  var highways = {};
+  var bboxes = [];
+  var preserve = {
+    'motorway': true,
+    'primary': true,
+    'secondary': true,
+    'tertiary': true,
+    'trunk': true,
+    'residential': true,
+    'unclassified': true,
+    'service': true
+  };
+
+  layer.features.forEach(function(val) {
+    if (preserve[val.properties.highway] && (val.geometry.type === 'LineString' || val.geometry.type === 'MultiLineString') && val.properties.layer === undefined) {
+      val.properties._osmlint = 'crossinghighways';
+      var bbox = turf.extent(val);
+      bbox.push(val.properties._osm_way_id);
+      bboxes.push(bbox);
+      highways[val.properties._osm_way_id] = val;
+    }
+  });
+
+  var traceTree = rbush(bboxes.length);
+  traceTree.load(bboxes);
+  var output = {};
+  var result = [];
+
+  bboxes.forEach(function(bbox) {
+    var overlaps = traceTree.search(bbox);
+    overlaps.forEach(function(overlap) {
+      if (bbox[4] !== overlap[4]) {
+        var intersect = turf.intersect(highways[overlap[4]], highways[bbox[4]]);
+        if (intersect !== undefined && (intersect.geometry.type === 'Point' || intersect.geometry.type === 'MultiPoint')) {
+          var highwaycoors = _.flatten(highways[overlap[4]].geometry.coordinates);
```

Comment:
@Rub21 Can you rename `highwaycoors` to `highwayCoord` to keep them uniform like [here](https://github.com/osmlab/osmlint/search?utf8=%E2%9C%93&q=Coord).